### PR TITLE
Add Encoding Support

### PR DESCRIPTION
**Pull request by zparis99**
Adds an encoding task, supporting code, and tests for data loading. Works on my local machine and would be interested to try on a real model. I've added a lot of TODO's which can be pointed to in the issues tab which I'll add tomorrow.

### CODE REVIEW

File: ECoG_MAE/downstream_tasks/encoding/config.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    
+    # The glob path to access electrode data within encoding_neural_data_folder. Should include placeholder for electrode_id {elec_id} See default for sample value.
+    # TODO: In the future include options for subject id and conv id but keeping it simple for now with just electrode id.
+    electrode_glob_path: str = "NY*_*_Part*_conversation*_electrode_preprocess_file_{elec_id}.mat"
+
+    # Determines where relative to the word onset in ms the neural signal should be read from.
+    # Lag = 0 means neural signal will be read starting at word onset.
+    # Lag = -2000 means neural signal will be read starting at 2 seconds before word onset.
+    # Lag = 100 means neural signal will be read starting at 100 ms before word onset.
+    lag: int = 0
```
Comment (itamarJa) [inline ECoG_MAE/downstream_tasks/encoding/config.py:21]: In the lab's pipeline, models for each electrode are normally trained for a series of lags [-2000:2000]. We need to decide how to think about this here for benchmarking

File: ECoG_MAE/downstream_tasks/encoding/load_signal.py [hunk 2]
```diff
@@ -0,0 +1,136 @@
+# Here we want to load the signal for the grid electrodes and organize it in an 8*8 matrix. We also want to perform filtering here - we might need to think about how to do it
+# such that it is the same as the filtering we perform for model training - since there we do it using mne, which we can't here, since the preprocessed signal here are mat files.
```
Comment (itamarJa) [inline ECoG_MAE/downstream_tasks/encoding/load_signal.py:2]: We can get the raw data and decide how we want to save (and process?) it. Do you have access to the raw edf file with this patient's podcast data?

**Review by itamarJa — APPROVED** [review]
This is great! Let me know what you would need to test this with actual training

File: ECoG_MAE/downstream_tasks/encoding/load_signal.py (hunk 2 above)
Comment (zparis99) [reply ECoG_MAE/downstream_tasks/encoding/load_signal.py:2]: I don't think I do currently. The way it's set up now is just using the 64 grid electrode data which Leo put on huggingface. If I had access to the edf file I could just use the existing data loader as well.

File: ECoG_MAE/downstream_tasks/encoding/config.py (hunk 1 above)
Comment (zparis99) [reply ECoG_MAE/downstream_tasks/encoding/config.py:21]: Good point, we can change this config to make this a list. I'm focusing on getting the model trained currently, so will add an issue for this to add support for this list.

File: ECoG_MAE/downstream_tasks/encoding/load_signal.py (hunk 2 above)
Comment (itamarJa) [reply ECoG_MAE/downstream_tasks/encoding/load_signal.py:2]: Great, maybe take a look at the current main branch in the encoding repo, it's been updated with more efficient regression (using himalaya)

**Pull request merged**